AAUnlimited/Functions/Notifications.cpp

```diff
@@ -0,0 +1,231 @@
+#include "StdAfx.h"
+
+namespace Notifications {
+
+	bool enabled = false;
+	std::wstring text;
+	std::list<std::tuple< std::wstring, int >> lines;
+	const int fontLayersCount = 9;
+	int outlineLayersCount = 8;		// Count Outline (black) layers to show for each Notifs lines (Depends outline quality)
+	RECT rect[fontLayersCount] = {	// 0 - shadow position | 0-7 - outline font pos | 8 - main font pos
+		{ 1, 41, 1024, 256 },{ -1, 41, 1024, 256 },
+	{ -1, 39, 1024, 256 },{ 1, 39, 1024, 256 },
+	{ -1, 40, 1024, 256 },{ 1, 40, 1024, 256 },
+	{ 0, 39, 1024, 256 },{ 0, 41, 1024, 256 },
+	{ 0, 40, 1024, 256 }
+	};
+	// 0 - outline/shadow | 1 - normal notify | 2 - important
+	D3DCOLOR colors[3] = { D3DCOLOR_RGBA(0, 0, 0, 255),
+		D3DCOLOR_RGBA(255, 190, 100, 255), D3DCOLOR_RGBA(255, 70, 25, 255) };
+
+	DWORD lastPopTime = 0;
+	int duration = 10000;
+	int maxLines = 4; // 4 for this var == 5 in game
+	const char *fontFamily = "Arial";
+	int fontSize = 18;
+	int lineHeight = 24;
+	IUnknown *Font;
+	int areaPosPercentsX = 0; // in `% * 100`
+	int areaPosPercentsY = 0;
+	int outlineSpread = 0;
+	int gameWindowWidth = 0;
+	int gameWindowHeight = 0;
+	int gameWindowMarginY = 0;
+	DWORD notifyCentered = 0;
+	bool separateColorType = true;
+
+	void AddNotification(const char *text, int type_id) {
+		if (!enabled) {
+			LOGPRIO(Logger::Priority::INFO) << text << "\r\n";
+			return;
+		}
+		if (type_id != 1) { type_id = 2; } // 1 - normal notify | 2 - important
+
+		if (lastPopTime == 0)
+			lastPopTime = GetTickCount();
+		if (lines.size() > maxLines)
+			lines.pop_front();
+		lines.push_back(std::make_tuple(General::utf8.from_bytes(text) + L"\n", type_id));
+	}
+
+	static int HexadecimalToDecimal(std::string hex) {
```

Comment:
Perhaps this should be moved to Utils.cpp for code sharing with subtitles.

Reply:
Ok